packages/kbn-es-query/index.ts

```diff
@@ -127,3 +130,21 @@ export {
 } from './src/utils';
 
 export type { ExecutionContextSearch } from './src/expressions/types';
+
+export function getKueryFields(nodes: KueryNode[]): string[] {
```

Comment:
It's a function I just copied from

https://github.com/elastic/kibana/blob/d228f488ec0456c96b3e06aee57a0d28af851eb4/x-pack/plugins/observability_solution/apm/common/utils/get_kuery_fields.ts#L11

it should be sufficient to extract fields from a KQL query

Reply:
Lukas is addressing this here - https://github.com/elastic/kibana/pull/183573